components/clipboard-button/index.js

```diff
@@ -78,13 +79,15 @@ class ClipboardButton extends Component {
 		// Disable reason: Exclude from spread props passed to Button
 		// eslint-disable-next-line no-unused-vars
 		const { className, children, onCopy, onFinishCopy, text, ...buttonProps } = this.props;
+		const { icon } = buttonProps;
 		const classes = classnames( 'components-clipboard-button', className );
+		const ComponentToUse = icon ? IconButton : Button;
```

Comment:
Maybe evidence that we could do for merging `IconButton` and `Button` (i.e. build `icon` behavior into `Button`). I know @mtias mentioned this on at least one occasion.